--- ./io.c
+++ ./io.c
@@ -8561,7 +8561,7 @@ do_ioctl(int fd, ioctl_req_t cmd, long narg)
 
 #define DEFULT_IOCTL_NARG_LEN (256)
 
-#ifdef __linux__
+#ifdef defined(__linux__) && defined(_IOC_SIZE)
 static long
 linux_iocparm_len(ioctl_req_t cmd)
 {
@@ -8594,7 +8594,7 @@ ioctl_narg_len(ioctl_req_t cmd)
 #endif
 #ifdef IOCPARM_LEN
     len = IOCPARM_LEN(cmd);	/* on BSDish systems we're safe */
-#elif defined(__linux__)
+#elif defined(__linux__) && defined(_IOC_SIZE)
     len = linux_iocparm_len(cmd);
 #else
     /* otherwise guess at what's safe */

--- ./missing/isnan.c
+++ ./missing/isnan.c
@@ -4,6 +4,7 @@
 
 static int double_ne(double n1, double n2);
 
+#ifndef isnan
 int
 isnan(double n)
 {
@@ -15,3 +16,4 @@ double_ne(double n1, double n2)
 {
     return n1 != n2;
 }
+#endif

--- ./missing/isinf.c
+++ ./missing/isinf.c
@@ -5,6 +5,7 @@
 #define _IEEE 1
 #include <nan.h>
 
+#ifndef isinf
 int
 isinf(double n)
 {
@@ -67,3 +68,4 @@ isinf(double n)
 }
 #endif
 #endif
+#endif
